### PR TITLE
Draft of workflow to explore/extend PR locally

### PR DESCRIPTION
**Pull request by jennybc**
WIP

cc @alexpghayes @jimhester

### CODE REVIEW

File: 36_workflow-explore-extend-pull-request.Rmd [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+)
+(sha <- purrr::pluck(y, list(1, "parents", 1, "sha")))
+
+## create and checkout the local PR branch, with correct parent
+b <- branch_create(commit = lookup(sha = sha), name = pr_branch)
+checkout(b)
+
+## haven't figured out how to pull / push to arbitrary remote with git2r
+## perhaps you'd have to actually add the fork as a remote?
+## doing with command line Git for now
```
Comment (jimhester) [inline 36_workflow-explore-extend-pull-request.Rmd]: It doesn't look like git2r has an API that lets you use anonymous remotes, I think you need to explicitly add the remote then fetch, then do the branch, e.g. `remote_add()`, `fetch()`, `branch()` etc.

File: 36_workflow-explore-extend-pull-request.Rmd [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+## alternative approach
+## GET all commits for the PR and parent SHA of the first commit
+y <- gh(
+  "/repos/:owner/:repo/pulls/:number/commits",
+  owner = repo_info$username,
+  repo = repo_info$repo,
+  number = pr
+)
+(sha <- purrr::pluck(y, list(1, "parents", 1, "sha")))
```
[the remark this answers is not in the document]
Comment (jennybc) [reply 36_workflow-explore-extend-pull-request.Rmd]: I don't understand this comment.

**Comment by jennybc** [comment]
Thanks @jimhester. I needed to reach temporary closure on this and go back to other tasks. But I incorporated your feedback as comments on the R code, so we don't lose it.

File: 36_workflow-explore-extend-pull-request.Rmd (hunk 2 above)
[the remark this answers is not in the document]
Comment (jimhester) [reply 36_workflow-explore-extend-pull-request.Rmd]: E.g. the difference between

```shell
git branch pr-123 abc123
```
Which is what you are doing here, where abc123 is the point on the master that the PR branched from, vs

```shell
git fetch a_remote a_branch
git branch pr-123 a_remote/a_branch
```

Which creates a local branch based on the remote branch.

[the remark this answers is not in the document]
Comment (jennybc) [reply 36_workflow-explore-extend-pull-request.Rmd]: Is this a more elegant way of getting the right topology or do you think the topology is different?

[the remark this answers is not in the document]
Comment (jennybc) [reply 36_workflow-explore-extend-pull-request.Rmd]: A question that is neither here nor there now but still bugs me. I cannot figure out the rationale for the `base` reported by [this API endpoint](https://developer.github.com/v3/pulls/#get-a-single-pull-request). Is it obvious to you?

[the remark this answers is not in the document]
Comment (jimhester) [reply 36_workflow-explore-extend-pull-request.Rmd]: The base pointed to there is what the PR is being compared against, e.g. in the compare dialog, which may not be the head of the current master depending on when the PR was opened.

**Pull request merged**